patches/chromium/parse_pkey_appusermodel_toastactivatorclsid_as_string_uuid.patch

```diff
@@ -0,0 +1,56 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?David=20L=C3=B6nnhager?= <dv.lnh.d@gmail.com>
+Date: Tue, 14 Jan 2025 09:36:44 +0100
+Subject: Parse PKEY_AppUserModel_ToastActivatorCLSID as string UUID
+
+Shortcuts may store the PKEY_AppUserModel_ToastActivatorCLSID property
+as a string UUID as opposed to VT_CLSID (i.e., bytes), resulting in
+NOTREACHED being hit. This was seen in the wild for a particular
+version of Slack. The CL adds support for VT_BSTR and VT_LPWSTR, and
+handles both cases with and without braces around the UUID.
+
+diff --git a/base/win/shortcut.cc b/base/win/shortcut.cc
+index e790adb2f1d6529ac0dd77145f5da2796264c7ae..b084ef302b411a45e6ce221a752852404f337925 100644
+--- a/base/win/shortcut.cc
++++ b/base/win/shortcut.cc
+@@ -4,6 +4,8 @@
+ 
+ #include "base/win/shortcut.h"
+ 
++#define RPC_USE_NATIVE_WCHAR
++
+ #include <objbase.h>
+ 
+ #include <propkey.h>
+@@ -341,6 +343,31 @@ bool ResolveShortcutProperties(const FilePath& shortcut_path,
+           properties->set_toast_activator_clsid(
+               *(pv_toast_activator_clsid.get().puuid));
+           break;
++        case VT_LPWSTR: {
++            CLSID clsid;
++            wchar_t *wstr = pv_toast_activator_clsid.get().pwszVal;
++            /* parse string of format "{...}" or "..." */
++            if (CLSIDFromString(wstr, &clsid) != NO_ERROR
++                && UuidFromStringW(wstr, &clsid) != NO_ERROR)
```

Comment:
Should this be `||` ? Same below.

Reply:
I should explain the point of each -- `UuidFromStringW` should be tried iff `CLSIDFromString` fails, because the latter doesn't handle IDs without curly braces.

So `CLSIDFromString(...) != NO_ERROR && UuidFromStringW(...) != NO_ERROR` seems correct

Reply:
Sorry, misread the error condition